Keep the symbolic link after editing credentials.yml.enc

sinsoku opened the pull request:
### Summary

If you edit credentials.yml.enc after creating a symbolic link,
the symbolic link is removed and overwritten with a normal file.
This commit changes the behavior to keep the symbolic link.

Fixes #36411

sinsoku committed:
Keep the symbolic link after editing credentials.yml.enc

If you edit credentials.yml.enc after creating a symbolic link,
the symbolic link is removed and overwritten with a normal file.
This commit changes the behavior to keep the symbolic link.

Fixes #36411

georgeclaghorn commented on activesupport/test/encrypted_file_test.rb:
```diff
@@ -56,4 +56,22 @@ class EncryptedFileTest < ActiveSupport::TestCase
+      )
+      link_encrypted_file.write(@content)
+
+      assert File.symlink?(link_path)
```
It’s worth asserting that the new contents were written to the realpath:

```ruby
assert_equal @content, @encrypted_file.read
```

sinsoku replied:
I'll add an assert to test writing to realpath.

kaspth commented on activesupport/lib/active_support/encrypted_file.rb:
```diff
@@ -48,7 +48,8 @@ def read
       IO.binwrite "#{content_path}.tmp", encrypt(contents)
-      FileUtils.mv "#{content_path}.tmp", content_path
+      path = content_path.symlink? ? content_path.realpath : content_path
+      FileUtils.mv "#{content_path}.tmp", path
```
Why do we put this check here versus when we assign content_path initially?

sinsoku replied:
Thank you for the review. I'll fix it.

sinsoku committed:
Create a new file when credentials.yml.enc is dead-symlink

An `Errno::ENOENT` occurs when using a symbolic link and the link destination does not exist.
This commit fixes to create a new file instead of causing an error.

sinsoku commented:
@georgeclaghorn @kaspth I pushed commits corresponding to the review.

* Check arguments with initialize
* Add an assert to test writing to realpath
* Rename "symbolic link" to "symlink" according to other files

Also I added a commit to fix an error when the symlink destination does not exist.

kaspth commented:
Edited the patch a bit for fun and merged in 678b59aa52, thanks so much! 😊🙏

sinsoku commented:
Your patch is so cool. Thank you! ❤️

pedrofurtado commented:
In which rails version this fix will be available?
